components/tag-list/tag-list.js

```diff
@@ -166,9 +168,10 @@ class TagList extends LocalizeCoreElement(ArrowKeysMixin(LitElement)) {
 
 			for (let i = overflowingIndex; i < this._itemLayouts.length; i++) {
 				const itemLayout = this._itemLayouts[i];
+				const itemWidth = Math.min(itemLayout.width, this._availableWidth);
 
-				if (!isOverflowing && showing.width + itemLayout.width < this._availableWidth) {
-					showing.width += itemLayout.width;
+				if (!isOverflowing && ((showing.width + itemWidth) <= (this._availableWidth + MARGIN_TOP_RIGHT))) {
```

Comment:
What's the purpose of the `MARGIN_TOP_RIGHT`?

Reply:
It corresponds to the 6px of top and right margin on tag items. I'd also had it used in the css but that was potentially an issue so I left it as a hard coded value there.